olio/service/resources/version_resource_test.go

```diff
@@ -0,0 +1,62 @@
+package resources_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/olioapps/service-skeleton-go/olio/service/resources"
+	"github.com/stretchr/testify/assert"
+)
+
+type VersionExtractor struct {
+}
+
+func NewVersionExtractor() *VersionExtractor {
+	return &VersionExtractor{}
+}
+
+func (ve VersionExtractor) GetVersion() string {
+	return "0.0.11"
+}
+
+func (ve VersionExtractor) GetAppName() string {
+	return "cx-messaging"
+}
+
+func TestVersion(t *testing.T) {
+	versionExtractor := NewVersionExtractor()
+
+	tt := []struct {
+		name string
+	}{
+		{name: "No extractor"},
+		{name: "With extractor"},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.name == "No extractor" {
+				router := gin.New()
+				versionResource := resources.NewVersionResource()
+				router.GET("/api/version", versionResource.GetVersion)
+				req, _ := http.NewRequest("GET", "/api/version", nil)
+				res := httptest.NewRecorder()
+				router.ServeHTTP(res, req)
+				assert.Equal(t, "{\n    \"serviceFrameworkVersion\": \""+resources.VERSION+"\",\n    \"appVersion\": \"no version given\"\n}", res.Body.String())
```

Comment:
"no version given" <--- is this being returned by the framework if there is no version extractor provided? if so, it shouldn't do that, it should just have no appVersion attribute